src/exts/start_new_spring_handler.js

```diff
@@ -0,0 +1,25 @@
+const { join } = require('path');
+
+const { bridge } = require('../spring_api');
+const springPlatform = require('../spring_platform');
+const launcher = require('../engine_launcher');
+
+bridge.on('StartNewSpring', async command => {
+	// Fetch the engine version
+	const enginePath = join(
+		springPlatform.writePath,
+		'engine',
+
+		// We need to put the engine's version in lowercase, because the
+		// directory name is in lowercase, and FS is case sensitive on
+		// some OSes.
+		command.Engine.toLowerCase(),
+
+		springPlatform.springBin
+	);
+
+	// Build the replay's path
+	const replayPath = join(springPlatform.writePath, 'demos', command.StartDemoName);
+
+	launcher.launchSpring(enginePath, [replayPath]);
```

Comment:
If we use the same global instance here, we risk causing unwanted behavior when demo has been watched.
For example, in main.js we'd end up closing the app once launcher finishes:
```
launcher.on('finished', (code) => {
	log.info(`Spring finished with code: ${code}`);
	app.quit();
	setTimeout(() => {
		gui.send('launch-finished');
	}, 100);
});
```

I recommend you make `engine_launcher` export the Launcher class and make a new object of this class here (on demand). You'd still want to do some basic logging like it's done in `main.js`. 

If you want to take it a step further and completely remove the instance from being exported in `engine_launcher`, please be sure that the two existing places that use it (`launcher_wizard.js` and `main.js`) are both working with the same instance.

Reply:
TBH if two files have to share an instance, it might aswell be declared in ̀engine_launcher.js`, at least as far as I understand the architecture, so I think I'll just keep the global instance and export the class.

Reply:
Honestly I think all these singletons should probably be defined in one file, but I won't require you to change this in this PR. (It's messy to export both class and instance, and the code is starting to lose consistency, some file export classes, other export instances)

> You'd still want to do some basic logging like it's done in main.js.
This is the only remaining change I'd like to see done. Right now you won't log anything from demos, which might make tracing issues more difficult

Reply:
> Honestly I think all these singletons should probably be defined in one file, but I won't require you to change this in this PR. (It's messy to export both class and instance, and the code is starting to lose consistency, some file export classes, other export instances)

Ok, I agree with what you say but I'd rather let you do this part, or do it in another PR if you give me precise instructions on what you want.

I'll add logging!